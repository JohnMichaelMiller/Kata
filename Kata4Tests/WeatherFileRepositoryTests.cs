using Microsoft.VisualStudio.TestTools.UnitTesting;
using Kata4;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using ApprovalTests;
using ApprovalTests.Reporters;

namespace Kata4.Tests
{
    [TestClass()]
    [UseReporter(typeof(DiffReporter))]
    public class WeatherFileRepositoryTests
    {
        [TestMethod()]
        public void WeatherFileRepositoryTest()
        {
            var repo = new WeatherFileRepository();
            Approvals.VerifyAll(repo, "Weather Data Point");
        }
    [TestMethod()]
    public void WeatherFileRepositoryHasCorrectMinimumTemperatureDifferential()
    {
        var repo = new WeatherFileRepository();
        var result =
            from w in repo
            where -w.TemperatureDifferential ==
                repo.Min(d => -d.TemperatureDifferential)
            select w;

        Approvals.VerifyAll(result, "Weather Data Point");
    }

    }
}