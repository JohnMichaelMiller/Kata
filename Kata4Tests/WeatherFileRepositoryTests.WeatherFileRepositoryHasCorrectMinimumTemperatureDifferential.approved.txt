Weather Data Point[0] = Day:14 Min Temp: 59, Max Temp 59, Temp Diff: -2
